wxt_cidrbot/git_webhook_handler.py

```diff
@@ -199,12 +223,83 @@ def triage_issue(self, installation_id, json_string, x_event_type):
+                room_message = reply_message[3]
+
+        self.Api.messages.create(room_id, markdown=room_message, parentId=msg_edit_id)
+
+    def send_codeowners_message(
+        self, issue, room_id, hyperlink_format, hyperlink_format_repo, issue_type, installation_id, json_string
+    ):
+        issue = issue.as_pull_request()
+        reviewers = issue.raw_data['requested_reviewers']
+
+        self.logging.debug("Codeowners is active, sending update message and quitting")
+        reviewer_message = ""
+        for reviewer in reviewers:
+            reviewer_message += reviewer['login'] + ', '
+        reviewer_message = reviewer_message[:-2]
+        empty_triage_message = f"{issue_type} {hyperlink_format} created in {hyperlink_format_repo}. This {issue_type} is auto assigned to {reviewer_message} via codeowners"
+        self.Api.messages.create(room_id, markdown=empty_triage_message)
+        self.logging.debug("Sending message to code owners")
+        self.send_review_message(installation_id, json_string, True, reviewers)
+        sys.exit(1)
+
+    def send_merged_message(self, installation_id, json_string):
+        event_info = self.check_installation(installation_id)
+        room_id = event_info[0]['room_id']
+
+        if json_string['pull_request']['merged'] is True:
+            issue_title = json_string['pull_request']['title']
+            issue_url = json_string['pull_request']['url']
+            issue_user = json_string['pull_request']['user']['login']
+            repo_name = json_string['repository']['full_name']
+            repo_url = json_string['repository']['html_url']
+
+            hyperlink_format = f'<a href="{issue_url}">{issue_title}</a>'
+            hyperlink_format_repo = f'<a href="{repo_url}">{repo_name}</a>'
+            merged_message = f"Pull request {hyperlink_format} has been merged in {hyperlink_format_repo} by {issue_user}"
+            self.Api.messages.create(room_id, markdown=merged_message)
+
+    def send_review_message(self, installation_id, json_string, codeowners_status, reviewers):
+        event_info = self.check_installation(installation_id)
+        room_id = event_info[0]['room_id']
+        assigned_reviewers = json_string['pull_request']['requested_reviewers']
+        reviewer_count = len(assigned_reviewers)
+        if reviewer_count > 1 and codeowners_status is False:
+            self.logging.debug("Codeowners will handle message sending, quitting")
+            sys.exit(1)
+
+        if codeowners_status:
+            assigned_reviewers = reviewers
+
+        for reviewer in assigned_reviewers:
+            try:
+                user = self.dynamo.get_user_info(reviewer['login'], room_id)
+                user_id = user['person_id']
+                user_name = user['first_name']
+            except Exception:
+                self.logging.debug("Cannot locate user, checking next...")
+                continue
```

Comment:
what happens here if you never find a user?

Reply:
Resolved in dynamo api handler to return None if it cannot find a user, and using an if else instead of try except